player/loadfile.c

```diff
@@ -1521,6 +1523,83 @@ static void load_external_opts(struct MPContext *mpctx)
     mp_waiter_wait(&wait);
 }
 
+static void append_to_watch_history(struct MPContext *mpctx)
+{
+    if (!mpctx->opts->save_watch_history)
+        return;
+
+    void *ctx = talloc_new(NULL);
+    char *history_path = mp_get_user_path(ctx, mpctx->global,
+                                          mpctx->opts->watch_history_path);
+    FILE *history_file = fopen(history_path, "ab");
```

Comment:
I was also thinking if we should use stream.c for this writing, instead of NIHing. Though stream.c doesn't support append currently, so it would need to be extended. And maybe for single write it is not that needed.